wpimath/src/main/native/include/frc/MathUtil.h

```diff
@@ -169,4 +169,39 @@ constexpr units::radian_t AngleModulus(units::radian_t angle) {
                                        units::radian_t{std::numbers::pi});
 }
 
+// floorDiv and floorMod algorithms taken from Java
+
+/**
+ * Returns the largest (closest to positive infinity)
+ * {@code int} value that is less than or equal to the algebraic quotient.
+ *
+ * @param x the dividend
+ * @param y the divisor
+ * @return the largest (closest to positive infinity)
+ *   {@code int} value that is less than or equal to the algebraic quotient.
+ */
+constexpr std::integral auto FloorDiv(std::integral auto x, std::integral auto y) {
+  auto quot = x / y;
+  auto rem = x % y;
+  // if the signs are different and modulo not zero, round down
+  if ((x ^ y) < 0 && rem != 0) {
```

Comment:
```suggestion
  if (std::signbit(x) != std::signbit(y) && rem != 0) {
```

Reply:
Hm, this causes all the tests to fail, despite it ostensibly doing what the comment says it should.